### PR TITLE
No Bug: Bump brave-core to 1.44.72

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Bump brave-core to 1.44.72 and related code updates.

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: App/iOS/Delegates/AppDelegate.swift [hunk 1]
```diff
@@ -52,6 +52,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         }
       }
     }
+    switches[.rewardsFlags] = BraveRewards.Configuration.current().flags
```
Comment (iccub) [inline App/iOS/Delegates/AppDelegate.swift:55]: Need @kylehickinson to approve this line

File: Client/Rewards/BraveRewards.swift [hunk 2]
```diff
@@ -262,43 +256,96 @@ class BraveRewards: NSObject {
...
+      switch environment {
+      case .production:
+        flags["staging"] = "false"
+      case .staging:
+        flags["staging"] = "true"
+      case .development:
+        flags["development"] = "true"
+      }
+      if let debug = isDebug {
+        flags["debug"] = debug ? "true" : "false"
```
Comment (iccub) [inline Client/Rewards/BraveRewards.swift]: Would `String(debug)` work here to cast bool to String?

**Review by iccub — APPROVED** [review]
Looks good. I would like to have Kyle to formally approve Rewards changes

File: App/iOS/Delegates/AppDelegate.swift (hunk 1 above)
Comment (kylehickinson) [reply App/iOS/Delegates/AppDelegate.swift:55]: @iccub These are my own changes lol

File: Client/Rewards/BraveRewards.swift (hunk 2 above)
Comment (kylehickinson) [reply Client/Rewards/BraveRewards.swift]: TIL there was an initializer on `String` that takes a `Bool`, yeah that would be better

Comment (nuo-xu) [reply Client/Rewards/BraveRewards.swift]: fixed in https://github.com/brave/brave-ios/pull/5960/commits/d16631c3cb38b8978f6c1d939b4ea15b7701ab43

**Review by kylehickinson — APPROVED** [review]

**Review by StephenHeaps — APPROVED** [review]

**Pull request merged**